crypto/ctype.c

```diff
@@ -278,3 +290,69 @@ int ossl_ascii_isdigit(const char inchar) {
         return 1;
     return 0;
 }
+
+/* str[n]casecmp_l is defined in POSIX 2008-01. Value is taken accordingly
+ * https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html */
+
+#if (defined OPENSSL_SYS_WINDOWS) || (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200809L)
+
+# if defined OPENSSL_SYS_WINDOWS
+# define locale_t _locale_t
+# define freelocale _free_locale
+# define strcasecmp_l _stricmp_l
+# define strncasecmp_l _strnicmp_l
+# endif
+static locale_t loc;
+
+# ifndef FIPS_MODULE
+void *OPENSSL_strcasecmp_locale() {
```

Comment:
Not sure I like that name much.  `OPENSSL_locale()` would be better, 'cause even if we implement that specifically for strcasecmp *now*, it's still a *locale* (unattributed), not a *strcasecmp locale*

Reply:
It's actually OPENSSL_C_locale().

Reply:
renamed to OPENSSL_C_locale. 

BTW, if we need to use it for something else, we will have to modify the mask (currently LC_COLLATE)